packages/@aws-cdk/core/lib/construct-compat.ts

```diff
@@ -226,16 +233,24 @@ export class ConstructNode {
    * @param options Synthesis options.
    */
   public static synth(root: ConstructNode, options: SynthesisOptions = { }): cxapi.CloudAssembly {
-    const builder = new cxapi.CloudAssemblyBuilder(options.outdir);
+    // We used to defer to `Node.synthesize()` here from the `constructs` library,
+    // but we don't do that anymore. Synthesis is deprecated from that library,
+    // and the implementation found there is not amenable to the kind of extension that we need.
```

Comment:
I'm not intimately familiar with the relationship between the construct in the CDK and in the `constructs` library. 

So far, I thought the `constructs` library has taken up all responsibility of maintaining the construct tree, tree operations and their order. It federates how certain nodes behave under operations like preparation and synthesis. 
Has this changed, or have I misunderstood? If this has changed, can you describe what the new relationship is?

> Synthesis is deprecated from that library

Don't intend to be pedantic but it doesn't say so - https://github.com/aws/constructs/blob/95b5fedc46eeea4e7835024a5403bc37f3b91479/lib/construct.ts#L382.
Somewhere else I've missed this?

Reply:
This is from a discussion between Elad and myself earlier this week, and this is my take-away. I'll let him confirm or deny.

Reply:
It's a directional thought, not a concrete plan yet. I am starting to realize that, similarly to "prepare", there is no huge value in a decentralized algorithm for synthesis. In fact, it just complicates things and makes it hard to implement certain features (for example, in cdk8s we need the names of the synthesized files to reflect a topological ordering like 015-foobar.k8s.yaml, and this knowledge is only centrally available at the app level).

My thinking is that, similarly to the refactor we did for app.prepare, we should do a similar refactor for app.synthesize. Basically move all the synthesis logic to the app/stage level.

@rix0rrr this is also why I noted that I wished this change will take this one step further and simply get rid of all synthesize methods. I don't think it's too far fetched to do that now and I am willing to pick this up if you prefer.